docs/guide/checking_nan.rst

```diff
@@ -0,0 +1,188 @@
+
+  print("numpy test:")
+
+  a = np.float64(1.0)
+  b = np.float64(0.0)
+  val = a / b  # this will now raise an exception instead of a warning.
+  print(val)
+
+but this will also avoid overflow issues on floating point numbers:
+
+.. code-block:: python
+
+  import numpy as np
+
+  np.seterr(all='raise')  # define before your code.
+
+  print("numpy overflow test:")
+
+  a = np.float64(10)
+  b = np.float64(1000)
+  val = a ** b  # this will now raise an exception
+  print(val)
+
+but will not avoid the propagation issues:
+
+.. code-block:: python
+
+  import numpy as np
+
+  np.seterr(all='raise')  # define before your code.
+
+  print("numpy propagation test:")
+
+  a = np.float64('NaN')
+  b = np.float64(1.0)
+  val = a + b  # this will neither warn nor raise anything
+  print(val)
+
+
+VecCheckNan Wrapper
+-------------------
+
+In order to find when and from where the invalid value originated from, Stable-baselines comes with a ``VecCheckNan`` wrapper.
+
+It will monitor the actions, observations, and rewards, indicating what action or observation caused it and from what.
+
+.. code-block:: python
+
+  import gym
+  from gym import spaces
+  import numpy as np
+
+  from stable_baselines import PPO2
+  from stable_baselines.common.vec_env import DummyVecEnv, VecCheckNan
+
+  class NanAndInfEnv(gym.Env):
+      """Custom Environment that raised NaNs and Infs"""
+      metadata = {'render.modes': ['human']}
+
+      def __init__(self):
+          super(NanAndInfEnv, self).__init__()
+          self.action_space = spaces.Box(low=-np.inf, high=np.inf, shape=(1,), dtype=np.float64)
+          self.observation_space = spaces.Box(low=-np.inf, high=np.inf, shape=(1,), dtype=np.float64)
+
+      def step(self, _action):
+          randf = np.random.rand()
+          if randf > 0.99:
+              obs = float('NaN')
+          elif randf > 0.98:
+              obs = float('inf')
+          else:
+              obs = randf
+          return [obs], 0.0, False, {}
+
+      def reset(self):
+          return [0.0]
+
+      def render(self, mode='human', close=False):
+          pass
+
+  # Create environment
+  env = DummyVecEnv([lambda: NanAndInfEnv()])
+  env = VecCheckNan(env, raise_exception=True)
+
+  # Instantiate the agent
+  model = PPO2('MlpPolicy', env)
+
+  # Train the agent
+  model.learn(total_timesteps=int(2e5))  # this will crash explaining that the invalid value originated from the environment.
+
+RL Model hyperparameters
+------------------------
+
+Depending on your hyperparameters, NaN can occurs much more often.
+A great example of this: https://github.com/hill-a/stable-baselines/issues/340
+
+Be aware, the hyperparameters given by default seem to work in most cases,
+however your environment might not play nice with the default hyperparameters.
+If this is the case, try to read up on the affect each hyperparameters has on the model,
+so that you can try and tune them to get a stable model.
```

Comment:
```suggestion
so that you can try and tune them to get a stable model. Alternatively, you can try automatic hyperparameter tuning (included in the rl zoo).
```